data/src/main/java/com/microsoft/azure/kusto/data/ClientImpl.java

```diff
@@ -41,18 +41,15 @@
     private final String userNameForTracing;
 
     public ClientImpl(ConnectionStringBuilder csb) throws URISyntaxException {
-        String url = csb.getClusterUrl();
-        URI clusterUri = new URI(url);
-        String host = clusterUri.getHost();
-        Objects.requireNonNull(clusterUri.getAuthority(), "clusterUri.authority");
-
-        String auth = clusterUri.getAuthority().toLowerCase();
+        URI clusterUrlForParsing = new URI(csb.getClusterUrl());
+        String host = clusterUrlForParsing.getHost();
+        Objects.requireNonNull(clusterUrlForParsing.getAuthority(), "clusterUri.authority");
+        String auth = clusterUrlForParsing.getAuthority().toLowerCase();
         if (host == null && auth.endsWith(FEDERATED_SECURITY_SUFFIX)) {
-            url = new URIBuilder().setScheme(clusterUri.getScheme()).setHost(auth.substring(0, clusterUri.getAuthority().indexOf(FEDERATED_SECURITY_SUFFIX))).toString();
-            csb.setClusterUrl(url);
+            csb.setClusterUrl(new URIBuilder().setScheme(clusterUrlForParsing.getScheme()).setHost(auth.substring(0, clusterUrlForParsing.getAuthority().indexOf(FEDERATED_SECURITY_SUFFIX))).toString());
```

Comment:
Editing the user's connection string is pretty unexpected
Is there a good reason to do it? do we do it anywhere else?

Reply:
I don't know the answer to either, unfortunately. My intent was just to make this code more readable.

Reply:
It seems changing the user's connection string was intentional, per https://github.com/Azure/azure-kusto-java/commit/c5079020c972903c61c8051b318ace921523bfec